data/json/scent_types.json

```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "type": "scent_type",
```

Comment:
Maybe you should also add a scent type for the standard scent (the normal human and non-zombie scent that every standard non-mutated character has).

Currently it seems you interpret an empty scent id as that type of scent and handle it specially. Having and actual scent for that case would make the code simpler as you don't have to handle special case.

Reply:
But then if a new species is added at some point that species will not be able to smell standard human if it's not added to the list, doesn't make the system more fragile?

Reply:
> But then if a new species is added at some point that species will not be able to smell standard human if it's not added to the list, doesn't make the system more fragile?

You're talking about a single word that needs to be added explicitly. This does not seem like a huge work.

Anyway. That argument could be made about the other scent types as well. If a new species is added, one has to consider *all* scent types anyway.

And how often is a species added anyway?

You could also added the date to the species itself (so each species lists explicitly what scents it does recognise).